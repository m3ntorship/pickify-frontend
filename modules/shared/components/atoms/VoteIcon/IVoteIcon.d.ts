declare namespace IVoteIcon {
  export interface IProps {
    isOneImageVote: boolean;
    like?: boolean;
    dislike?: boolean;
  }
}

export { IVoteIcon };
