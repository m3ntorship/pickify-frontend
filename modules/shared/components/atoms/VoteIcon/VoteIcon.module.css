.love,
.like,
.dislike {
  @apply outline-none bg-primary hover:ring-4 transition-all w-10 h-10 focus:outline-none transform hover:scale-145 flex justify-center items-center rounded-full cursor-pointer;
}

.like {
  @apply ml-4 hover:ml-8;
}

.dislike {
  @apply bg-error hover:mr-4 ring-error ring-opacity-25;
}
