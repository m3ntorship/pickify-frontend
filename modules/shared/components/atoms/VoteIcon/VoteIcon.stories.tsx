import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import VoteIcon from './VoteIcon';
import type { IVoteIcon } from './IVoteIcon';

export default {
  title: 'Atoms/VoteIcon',
  component: VoteIcon,
  argTypes: {
    isOneImageVote: {
      control: {
        type: 'boolean',
      },
    },
  },
};

const Template: Story<IVoteIcon.IProps> = (args): ReactElement => (
  <VoteIcon {...args} />
);

export const VoteIconStory = Template.bind({});
VoteIconStory.args = {
  isOneImageVote: false,
};
