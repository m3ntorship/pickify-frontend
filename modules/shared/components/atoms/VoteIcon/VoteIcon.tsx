import type { ReactElement, FC } from 'react';
import React from 'react';
import type { IVoteIcon } from './IVoteIcon';
import FilledHeartIcon from '../../icons/filledHeart.svg';
import FilledDislikeIcon from '../../icons/filledDislike.svg';
import FilledLikeIcon from '../../icons/filledLike.svg';
import styles from './VoteIcon.module.css';

const VoteIcon: FC<IVoteIcon.IProps> = ({ isOneImageVote }): ReactElement => {
  return (
    <div className="flex justify-end">
      {!isOneImageVote && (
        <button type="button" className={`${styles.love} group`}>
          <FilledHeartIcon className="fill-primary stroke-current stroke-2 text-white group-hover:fill-white" />
        </button>
      )}
      {isOneImageVote && (
        <div className="flex">
          <button type="button" className={`${styles.dislike} group`}>
            <FilledDislikeIcon className="fill-error stroke-current stroke-1 text-white group-hover:fill-white" />
          </button>
          <button type="button" className={`${styles.like} group`}>
            <FilledLikeIcon className="fill-primary stroke-current stroke-1 text-white group-hover:fill-white" />
          </button>
        </div>
      )}
    </div>
  );
};

export default VoteIcon;
