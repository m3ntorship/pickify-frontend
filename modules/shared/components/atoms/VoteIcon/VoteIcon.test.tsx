import React from 'react';
import renderer from 'react-test-renderer';
import VoteIcon from './VoteIcon';

describe('snapshot testing', () => {
  it('should render heart button when oneImage is true', () => {
    const tree = renderer.create(<VoteIcon isOneImageVote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render two buttons when oneImage is true', () => {
    const tree = renderer.create(<VoteIcon isOneImageVote={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
