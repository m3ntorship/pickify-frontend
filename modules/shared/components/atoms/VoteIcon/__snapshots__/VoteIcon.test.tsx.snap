// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing should render heart button when oneImage is true 1`] = `
<div
  className="flex justify-center"
>
  <div
    className="flex"
  >
    <button
      className="dislike group"
      type="button"
    >
      <svg
        className="fill-error stroke-current stroke-1 text-white group-hover:fill-white"
        data-file-name="SvgFilledDislike"
      />
    </button>
    <button
      className="like group"
      type="button"
    >
      <svg
        className="fill-primary stroke-current stroke-1 text-white group-hover:fill-white"
        data-file-name="SvgFilledLike"
      />
    </button>
  </div>
</div>
`;

exports[`snapshot testing should render two buttons when oneImage is true 1`] = `
<div
  className="flex justify-center"
>
  <button
    className="love group"
    type="button"
  >
    <svg
      className="fill-primary stroke-current stroke-2 text-white group-hover:fill-white"
      data-file-name="SvgFilledHeart"
    />
  </button>
</div>
`;
